ReactWindows/ReactNative.Shared/UIManager/ReactShadowNode.cs

```diff
@@ -1191,18 +1212,38 @@ public int GetNativeOffsetForChild(ReactShadowNode child)
             return index;
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
         /// <summary>
         /// Disposes the shadow node.
         /// </summary>
-        public void Dispose()
+        /// <param name="disposing">
+        ///     <code>true</code> if disposing directly,
+        ///     <code>false</code> if called by finalizer.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
         {
+            if (_disposed)
+            {
+                return;
+            }
+
             YogaNode nodeToDispose = (IsDelegatedLayout && _initialYogaNode != null) ? _initialYogaNode : _yogaNode;
 
             if (nodeToDispose != null)
             {
+                RemoveAndDisposeAllChildren();
+                UnlinkFromParent();
```

Comment:
I guess my only other concern is why this is causing a memory leak here but not on Android? The dispose method in ReactShadowNodeImpl on Android doesn't call either of these, is there a memory leak there, or are we missing a larger issue with the processing of lifecycle events?

Under what conditions does the memory leak happen?